api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -139,9 +139,11 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 							Obs replacement = os.saveObs(o, changeMessage);
 							//The logic in saveObs evicts the old obs instance, so we need to update
 							//the collection with the newly loaded and voided instance
-							toRemove.add(o);
-							toAdd.add(os.getObs(o.getId()));
-							toAdd.add(replacement);
+							if(!replacement.equals(o)) {
```

Comment:
In theory, my take is that if you add a new member or remove an existing one from a parent, the parent should  be marked as dirty. On the other hand, editing a child without changes to the parent should not make the parent dirty  even though technically we replace the voided obs with the new one. May be we should ask this on talk and see what others think